java/Cherepnin_Oleh/src/com/zoolatech/lecture5/tasks/_4/TaskFour.java

```diff
@@ -0,0 +1,41 @@
+package com.zoolatech.lecture5.tasks._4;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Create a method that accepts a string, counts all unique non-space character occurrences and displays each character
+ * with an amount of occurrences (output should be sorted by a character value in ascending order).
+ * Also, add a couple of examples that use your method.
+ * Input: “Hello World”
+ * Output: H - 1
+ * W - 1
+ * d - 1
+ * e - 1
+ * l - 3
+ * o - 2
+ * r - 1
+ */
+public class TaskFour {
+    public static void main(String[] args) {
+        countCharacters("Hello World");
+        System.out.println("-----------------");
+        countCharacters("Aloha from Hawaii");
+        System.out.println("-----------------");
+        countCharacters("Good evening");
+    }
+
+    static void countCharacters(String string) {
+        Map<Character, Integer> map = new TreeMap<>();
+        char[] chars = string.replaceAll(" ", "").toCharArray();
```

Comment:
instead of replacing characters in the string (which uses, by the way, a regular expression, which is a heavy operation) and creating a new string, you could simply add an if statement which will skip whitespace chars